src/main/java/com/google/cloud/spanner/pgadapter/utils/MutationWriter.java

```diff
@@ -257,10 +257,9 @@ public StatementResult call() throws Exception {
         }
         if (record.numColumns() != this.tableColumns.keySet().size()) {
           throw PGExceptionFactory.newPGException(
-              "Invalid COPY data: Row length mismatched. Expected "
-                  + this.tableColumns.keySet().size()
-                  + " columns, but only found "
-                  + record.numColumns(),
+              String.format(
```

Comment:
This change is not directly related to the addition of `float4`, but it popped up during the development that the error message was a bit weird. The error is thrown when the number of columns in the input of the COPY operation is different from the expected number of columns. The error message 'assumed' that the number of input columns was always smaller than the expected number of values (see the '... but only found' part).